packages/core/utils/src/dml/entity.ts

```diff
@@ -7,5 +12,39 @@ import { RelationshipType, SchemaType } from "./types"
 export class DmlEntity<
   Schema extends Record<string, SchemaType<any> | RelationshipType<any>>
 > {
+  #hooks: EntityHooks = {}
   constructor(public name: string, public schema: Schema) {}
+
+  /**
+   * Parse entity to get its underlying information
+   */
+  parse(): {
+    name: string
+    schema: SchemaType<any> | RelationshipType<any>
+    hooks: EntityHooks
+  } {
+    return {
+      name: this.name,
+      schema: this.schema as unknown as SchemaType<any> | RelationshipType<any>,
+      hooks: this.#hooks,
+    }
+  }
+
+  /**
+   * Delete actions to be performed when the entity is deleted. For example:
+   *
+   * You can configure relationship data to be deleted when the current
+   * entity is deleted.
+   */
+  onDelete(options: { remove: ExtractEntityRelations<Schema> }) {
```

Comment:
I am just thinking if `onDelete` is the best name or not. (this is especially true for the `onCreate` hook, is it before or after creation? It isn't obvious).

Other options:
`onDeleted`
`deleted`
`afterDelete`


What does everyone think? @olivermrbl @adrien2p @srindom

Reply:
yes indeed, I think the issue here is that we are using the `onDelete` hook name to configure an option related to a cascade which occurs upon true deletion. if you agree, when you are deleting something on cascade, the child must be deleted first and therefore we could assume that it is before, but then let say we open the configuration of hooks to beforeDelete, afterDelete now we can see that configuring a cascade cannot necessarely be represented by the hook configuration.

Cant we separate the two and have the onDelete where we configure the cascade and later on add the notion of `hooks` where you can configure this before/after actions?

Reply:
I don't have any strong opinion on this. Think `afterDelete` is also a good option. If we ever want to add a hook for something that happens before, then the explicitness makes the distinction between the two obvious.

Reply:
so it would mean that after delete would be able to take the configuration for cascading plus hooks and before delete only for the hooks? doesnt that seems strange?

Reply:
I agree on the thinking no worries I am the one that keep repeating it 😂 , I am thinking more the information we want to provide to the user, which is indeed a cascade information that happen when a tuple gets deleted 😂 what I am thinking is that `onDelete` could be the one taking the casacade configuration and all hooks would be `(before/after)Action` that way there is a clear distinction and it is also impossible to have a hook during deletion wdyt?